### PR TITLE
Various improvements related to segments

### PR DESCRIPTION
**Pull request by mtyaka**
This PR implements the following enhancements:

* Allow to set the color of control points at the ends of vectors/segments
* For segments, make the tail of each line draggable.
* Allow course authors to disable the "vector properties" display
* Allow to set the header in the vector properties (for eg. to "Curve Properties")
* Allow course authors to provide custom error messages for each check.

### CODE REVIEW

**Comment by mtyaka** [comment]
@e-kolpakov Please review.

File: README.md [hunk 1]
```diff
@@ -137,6 +143,33 @@ The `tail_tolerance` and `tip_tolerance` define the maximum allowed
...
 coordinates.
 
+Each property check comes with a default error message that can be
+overriden by specifying a `<property>_errmsg` entry. For example, if
+you want the error message of a `tail_x` check to be something other
+than the default `"Vector N does not start at the correct point"`, you
+would specify the expected result like this:
+
+    expected_result: {
+        N: {tail_x: 5, tail_errmsg: 'N starts at a wrong location, try again.'}
```
Comment (e-kolpakov) [inline README.md]: @mtyaka `tail_errmsg` or `tail_x_errmsg`?

**Comment by e-kolpakov** [comment]
@mtyaka :+1:

File: README.md (hunk 1 above)
Comment (mtyaka) [inline README.md]: Oops.... fixed, thanks.

**Comment by antoviaque** [comment]
@dseaton FYI - the set of changes to segment support from this sprint. Don't hesitate if you have comments or issues with it.

**Comment by dseaton** [comment]
@antoviaque @mtyaka This pull request looked good to me. I was able to make new examples using the functionality. Thanks!

**Pull request merged**